pkg/deploy/registry/deployconfig/etcd/etcd.go

```diff
@@ -46,3 +73,157 @@ func NewStorage(s storage.Interface) *REST {
+		return nil, false, errors.NewBadRequest(fmt.Sprintf("nil update passed to Scale"))
+	}
+	scale, ok := obj.(*extensions.Scale)
+	if !ok {
+		return nil, false, errors.NewBadRequest(fmt.Sprintf("wrong object passed to Scale update: %v", obj))
+	}
+	deployment, err := (*r.registry).GetDeploymentConfig(ctx, scale.Name)
+	if err != nil {
+		return nil, false, errors.NewNotFound("scale", scale.Name)
+	}
+
+	useDC := false
+
+	// this tries to update the current deployment RC first, since updating
+	// Replicas on a DeploymentConfig doesn't do anything after the first deployment
+	// has been made
+	controller, err := r.rcNamespacer.ReplicationControllers(deployment.Namespace).Get(util.LatestDeploymentNameForConfig(deployment))
+	if err != nil {
+		if !errors.IsNotFound(err) {
+			return nil, false, err
+		}
+
+		useDC = true
+	}
+
+	if isDeploymentInflight(controller) {
+		return nil, false, errors.NewConflict("Scale", scale.Name, fmt.Errorf("deployment currently in progress"))
+	}
+
+	// TODO(directxman12): this is going to be a bit out of sync, since we are calculating it
+	// here and not as part of the deploymentconfig loop -- is there a better way of doing it?
+	totalReplicas, err := r.replicasForDeploymentConfig(deployment.Namespace, deployment.Name)
+	if err != nil {
+		return nil, false, err
+	}
+
+	if useDC {
+		deployment.Template.ControllerTemplate.Replicas = scale.Spec.Replicas
+		if err = (*r.registry).UpdateDeploymentConfig(ctx, deployment); err != nil {
+			return nil, false, errors.NewConflict("scale", scale.Name, err)
+		}
+	} else {
+		oldReplicas := controller.Spec.Replicas
+		controller.Spec.Replicas = scale.Spec.Replicas
+		if _, err = r.rcNamespacer.ReplicationControllers(deployment.Namespace).Update(controller); err != nil {
+			return nil, false, errors.NewConflict("scale", scale.Name, err)
+		}
+		totalReplicas += (scale.Spec.Replicas - oldReplicas)
+	}
+
+	return &extensions.Scale{
+		ObjectMeta: kapi.ObjectMeta{
+			Name:              deployment.Name,
+			Namespace:         deployment.Namespace,
+			CreationTimestamp: deployment.CreationTimestamp,
+		},
+		Spec: extensions.ScaleSpec{
+			Replicas: scale.Spec.Replicas,
+		},
+		Status: extensions.ScaleStatus{
+			Replicas: totalReplicas,
+			Selector: deployment.Template.ControllerTemplate.Selector,
+		},
+	}, false, nil
+}
+
+func (r *ScaleREST) deploymentsForConfig(namespace, configName string) (*kapi.ReplicationControllerList, error) {
+	selector := util.ConfigSelector(configName)
+	return r.rcNamespacer.ReplicationControllers(namespace).List(selector, fields.Everything())
+}
+
+func (r *ScaleREST) replicasForDeploymentConfig(namespace, configName string) (int, error) {
+	rcList, err := r.deploymentsForConfig(namespace, configName)
+	if err != nil {
+		return 0, err
+	}
+
+	replicas := 0
+	for _, rc := range rcList.Items {
+		replicas += rc.Spec.Replicas
+	}
+
+	return replicas, nil
+}
+
+func isDeploymentInflight(deployment *kapi.ReplicationController) bool {
```

Comment:
There is already an utility for this: https://github.com/openshift/origin/blob/46d8271a97de24c07464198d28694eb4dd0072d7/pkg/deploy/util/util.go#L261